apps/extension/src/ui/domains/Sign/Ethereum/shared/SignParamAllowanceButton.tsx

```diff
@@ -59,13 +62,33 @@ const isValidAmount =
     }
   }
 
+const useErc20Balance = (account: EvmAddress, token: Erc20TokenInfo) => {
```

Comment:
It would be more ideal if this query would look in the balances pool first to see if there's a balance for the token before making the RPC call, is that possible?
If not, might be best to rename the hook (eg `useFetchErc20Balance`) because devs could get confused and assume this uses a balance already held locally

Reply:
Hook renamed.
Kept it simple with just a contract balance fetch, because we dont want the whole balance subscriptions to start on the approval popup.
If we were to use balance libraries (which would do the same query, but with more overhead), we'd still need this contract call here as fallback in case the token is not registered in chaindata.